samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,365 @@
+        // First, flip target if both qubits are in |0⟩ state
+        (ControlledOnInt(0, X))(queryRegister, target);
+        // Then flip target again to get negation
+        X(target);
+    }
+    
+    /// # Summary
+    /// Using Grover's search to find vertex coloring.
+    ///
+    /// # Input
+    /// ## V
+    /// The number of Vertices in the graph
+    /// ## K
+    /// The number of bits per color
+    /// ## maxIterations
+    /// An estimate of the maximum iterations needed
+    /// ## oracle
+    /// The Oracle used to find solution
+    operation GroversAlgorithm (V : Int, K : Int, maxIterations : Int, 
+        oracle : ((Qubit[], Qubit) => Unit is Adj)) : Int[] {
+        // This task is similar to task 2.2 from SolveSATWithGrover kata, 
+        // but the percentage of correct solutions is potentially higher.
+        mutable coloring = new Int[V];
+
+        // Note that coloring register has the number of qubits that is 
+        // twice the number of vertices (K qubits per vertex).
+        using ((register, output) = (Qubit[K * V], Qubit())) {
+            mutable correct = false;
+            mutable iter = 1;
+            repeat {
+                Message($"Trying search with {iter} iterations");
+                GroversAlgorithm_Loop(register, oracle, iter);
+                let res = MultiM(register);
+                // to check whether the result is correct, apply the oracle to the 
+                // register plus ancilla after measurement
+                oracle(register, output);
+                if (MResetZ(output) == One) {
+                    set correct = true;
+                    // Read off coloring
+                    set coloring = MeasureColoring(K, register);
+                }
+                ResetAll(register);
+            } until (correct or iter > maxIterations)  
+            fixup {
+                set iter += 1;
+            }
+            if (not correct) {
+                fail "Failed to find a coloring";
+            }
+        }
+        return coloring;
+    }
+
+    /// # Summary
+    /// Grover loop implementation taken from SolveSATWithGrover kata.
+    operation OracleConverterImpl (markingOracle : ((Qubit[], Qubit) => Unit is Adj), 
+        register : Qubit[]) : Unit is Adj {
+
+        using (target = Qubit()) {
+            within {
+                // Put the target into the |-⟩ state
+                X(target);
+                H(target);
+            } apply {
+                // Apply the marking oracle; since the target is in the |-⟩ state,
+                // flipping the target if the register satisfies the oracle condition 
+                // will apply a -1 factor to the state
+                markingOracle(register, target);
+            }
+            // We put the target back into |0⟩ so we can return it
+        }
+    }
+    
+    /// # Summary
+    /// Grover's Algorithm loop
+    operation GroversAlgorithm_Loop (register : Qubit[], 
```

Comment:
As mentioned in earlier comment, I would remove the underscore from this operation name.